sympy/stats/frv_types.py

```diff
@@ -174,14 +174,9 @@ def Die(name, sides=6):
 class BernoulliDistribution(SingleFiniteDistribution):
     _argnames = ('p', 'succ', 'fail')
 
-    def __new__(cls, *args):
-        p = args[BernoulliDistribution._argnames.index('p')]
-        p_sym = sympify(p)
-
-        if fuzzy_not(fuzzy_and((p_sym.is_nonnegative, (p_sym - 1).is_nonpositive))):
-            raise ValueError("p = %s is not in range [0, 1]." % str(p))
-        else:
-            return super(BernoulliDistribution, cls).__new__(cls, *args)
+    @staticmethod
+    def check(p, succ, fail):
+        _value_check(And(p >= 0, p <= 1), "p should be in range [0, 1].")
```

Comment:
careful, this may not be convertable into boolean.

Reply:
@Upabjojr This logic is used in drv_types too. An example will be helpful, so that changes can be made there as well.
Should I use `fuzzy_not` and `fuzzy_and`?

Reply:
This is the [line](https://github.com/sympy/sympy/blob/master/sympy/stats/drv_types.py#L46) I was referring too. Changes are to be made here too if `And(p >= 0, p <= 1)` is not convertible to boolean.

Reply:
@Upabjojr What are your views on the above the comments?

Reply:
@Upabjojr Any updates on this? Waiting for your comments.

Reply:
@Upabjojr I have updated the checks, kindly take a look and tell me if its fine.